data/src/main/java/com/sacramento/data/datasource/MoviesWithDetailsPagingSourceDB.kt

```diff
@@ -0,0 +1,72 @@
+package com.sacramento.data.datasource
+
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import com.sacramento.data.cache.MovieCategories
+import com.sacramento.data.cache.SharedPrefMovieCategory
+import com.sacramento.data.cache.SharedPrefMovieFilter
+import com.sacramento.data.utils.START_PAGE
+import com.sacramento.domain.models.MovieWithDetailsModel
+import com.sacramento.domain.repositories.MovieDBRepository
+import com.sacramento.domain.repositories.MoviesRepository
+import java.io.IOException
+
+class MoviesWithDetailsPagingSourceDB(
+    private val movieDBRepository: MovieDBRepository,
+    private val sharedPrefMovieCategory: SharedPrefMovieCategory,
+    private val shardPrefMovieFilter: SharedPrefMovieFilter
+) : PagingSource<Int, MovieWithDetailsModel>() {
+
+    override suspend fun load(params: LoadParams<Int>): LoadResult<Int, MovieWithDetailsModel> {
+        val page = params.key ?: START_PAGE
+        val moviesWithDetailsList = mutableListOf<MovieWithDetailsModel>()
+        val movieCategory = sharedPrefMovieCategory.loadMovieCategory()
```

Comment:
Is it possible to pass all these filter parameters into constructor as a some FilterParams data class?
If PagingSource classes know about SharedPrefs we get coupling that we could avoid.

Reply:
Fixed